### DISCUSSION TITLE
Use different Parameter in constructor for loading different Service (dependency injection)

### DISCUSSION THREAD

**Discussion by alessandro-podo** [symfony/symfony#43547] [Q&A] [answered]
I have a question about the Bundels.
With the monolog bundle I can load different channels due to the parameter in the constuctor.
How does this work?

**Comment by ahundiak** [comment, marked as answer]
Are you asking about [named autowiring aliases](https://symfony.com/doc/current/service_container/autowiring.html#dealing-with-multiple-implementations-of-the-same-type)?  Or something specifically about the monolog bundle?

**Reply by alessandro-podo** [reply]
This i find too, but i dint find this kind of configuration in the monolog-bundle. 

Is it possible, that they load dynamic the new Channels with this CompilerPass for the named aliases?
https://github.com/symfony/monolog-bundle/blob/master/DependencyInjection/Compiler/LoggerChannelPass.php
